pkg/cli/config/config.go

```diff
@@ -351,3 +352,56 @@ func getRawKubeConfigLoader(kubeconfigPath string) clientcmd.ClientConfig {
 	// Return the clientcmd.ClientConfig (equivalent to ToRawKubeConfigLoader)
 	return clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
 }
+
+// ServerConnectionFlagsChanged returns true if any server connection flags are set.
+func ServerConnectionFlagsChanged(cmd *cobra.Command) bool {
```

Comment:
ServerConnectionFlagsChanged and the following BuildDirectClientConfig are not used from outside the package. Do they have to be exported? Maybe if we add tests for them which we should probably do, then it makes sense.